website/source/docs/guides/acl-migrate-tokens.html.md

```diff
@@ -0,0 +1,415 @@
+running Vault 1.0.0 or higher, _and_ you must update all roles defined in Vault
+to specify a list of policy names rather than an inline policy (which causes
+Vault to use the legacy API).
+
+~> **Note:** if you have systems still creating "legacy" tokens with the old
+APIs, the migration steps below will still work, however you'll have to keep
+re-running them until nothing is creating legacy tokens to unsure all tokens are
+migrated.
+
+### Creating Policies
+
+There are several ways to create new policies. The simplest and most automatic
+is to create one new policy for every existing token. This may however result in
+a lot of policies that are logical duplicates and make managing policies harder.
+This approach can be easily accomplished using the [`consul acl policy
+create`](/docs/commands/acl/acl-policy.html#create) command with `-from-token`
+option.
+
+An alternative is to create one policy for each logically distinct token. While
+it's easy enough to do this by hashing the policy contents and only keeping
+distinct hashes, it's hard to extract a meaningful name for the policy that
+expresses it's intent. To make this easier, there is a CLI and API tool that can
+translate a legacy ACL policy into a new ACL policy that is exactly equivalent.
+See [`consul acl translate-rules`](/docs/commands/acl/acl-translate-rules.html).
+
+The final option is to manually inspect all the existing tokens and define named
+policies that describe what the policy is intended to be used for. In this case
+you may also want to modify the policy for example switching to exact-match
+rather than prefix match on resources if that is all that is actually required.
+In this case existing ACL token rules can be inspected using the [`consul acl
+token read -id <accessor_id>`](/docs/commands/acl/acl-token.html#read) command.
+
+### Updating Existing Tokens
+
+Once you have one or more policies that adequately express the rules needed for
+a legacy token, you can update the token via the CLI or API to use those
+policies.
+
+After updating, the token is no longer considered "legacy" and will have all the
+properties of a new token, however it keeps it's `SecretID` (the secret part of
+the token used in API calls) so clients already using that token will continue
+to work. It is assumed that the policies you attach continue to grant the
+necessary access for existing clients; this is up to the operator to ensure.
+
+#### Update via API
+
+Use the [`PUT /v1/acl/token/:AccessorID`](/api/acl/tokens.html#update-a-token)
+endpoint. Specifically, ensure that the `Rules` field is omitted or empty. Empty
+`Rules` indicates that this is now treated as a new token.
+
+#### Update via CLI
+
+Use the [`consul acl token update`](/docs/commands/acl/acl-token.html#update)
+command to update the token. Specifically you need to use `-upgrade-legacy`
+which will ensure that legacy rules are removed as well as the new policies
+added.
+
+## Migration Examples
+
```

Comment:
I was a little confused about the examples at first, I didn't know what to expect. A short intro paragraph would be nice. Something like:

"Below are several code examples that you can use as a model for your upgrade process. There are two strategies outlined; simple policy mapping and combining policies. Finally, there are examples of cleaning up policies, creating policies, and upgrading tokens."

Reply:
Yeah the intro is good. I had one but then kept changeing scope so jsut removed it. Also though I got the heading nesting wrong which explains your confusion. There are just two sub-sections here and all the headings under "Combining Policies" should be sub-headings under that. Just fixing.